File: BankingSystem.py
class Account:
    def __init__(self, account_number, owner_name, balance):
        self.account_number = account_number
        self.owner_name = owner_name
        self.balance = balance

    def deposit(self, amount):
        if amount > 0:
            self.balance += amount
            return f"Deposited ${amount}. New balance: ${self.balance}"
        else:
            return "Invalid deposit amount."

    def withdraw(self, amount):
        if amount > 0 and amount <= self.balance:
            self.balance -= amount
            return f"Withdrew ${amount}. New balance: ${self.balance}"
        else:
            return "Invalid withdrawal amount or insufficient balance."

    def get_balance(self):
        return f"Account balance for {self.owner_name}: ${self.balance}"

    def __str__(self):
        return f"Account Number: {self.account_number}\nOwner: {self.owner_name}\nBalance: ${self.balance}"


class Bank:
    def __init__(self):
        self.accounts = {}

    def create_account(self, account_number, owner_name, initial_balance):
        if account_number not in self.accounts:
            account = Account(account_number, owner_name, initial_balance)
            self.accounts[account_number] = account
            return f"Account created for {owner_name} with account number {account_number}."
        else:
            return "Account with that account number already exists."

    def get_account(self, account_number):
        if account_number in self.accounts:
            return str(self.accounts[account_number])
        else:
            return "Account not found."

    def deposit(self, account_number, amount):
        if account_number in self.accounts:
            return self.accounts[account_number].deposit(amount)
        else:
            return "Account not found."

    def withdraw(self, account_number, amount):
        if account_number in self.accounts:
            return self.accounts[account_number].withdraw(amount)
        else:
            return "Account not found."

    def get_balance(self, account_number):
        if account_number in self.accounts:
            return self.accounts[account_number].get_balance()
        else:
            return "Account not found."

def main():
    bank = Bank()
    while True:
        print("\nBanking Management System")
        print("1. Create Account")
        print("2. Get Account Info")
        print("3. Deposit")
        print("4. Withdraw")
        print("5. Quit")
        choice = input("Enter your choice: ")

        if choice == "1":
            account_number = input("Enter account number: ")
            owner_name = input("Enter owner's name: ")
            initial_balance = float(input("Enter initial balance: "))
            result = bank.create_account(account_number, owner_name, initial_balance)
            print(result)

        elif choice == "2":
            account_number = input("Enter account number: ")
            print(bank.get_account(account_number))

        elif choice == "3":
            account_number = input("Enter account number: ")
            amount = float(input("Enter the amount to deposit: "))
            result = bank.deposit(account_number, amount)
            print(result)

        elif choice == "4":
            account_number = input("Enter account number: ")
            amount = float(input("Enter the amount to withdraw: "))
            result = bank.withdraw(account_number, amount)
            print(result)

        elif choice == "5":
            print("Goodbye!")
            break

        else:
            print("Invalid choice. Please try again.")

if __name__ == "__main__":
    main()

File: BankingSystem.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Structure to store account information
struct Account {
    int accountNumber;
    char name[50];
    double balance;
};

// Function to create a new account
void createAccount(struct Account accounts[], int *numAccounts) {
    if (*numAccounts < 100) {
        struct Account newAccount;
        newAccount.accountNumber = *numAccounts + 1;

        printf("Enter account holder's name: ");
        scanf("%s", newAccount.name);

        printf("Enter initial balance: ");
        scanf("%lf", &newAccount.balance);

        accounts[*numAccounts] = newAccount;
        (*numAccounts)++;
        printf("Account created successfully.\n");
    } else {
        printf("Maximum account limit reached.\n");
    }
}

// Function to deposit money
void deposit(struct Account accounts[], int numAccounts, int accountNumber, double amount) {
    for (int i = 0; i < numAccounts; i++) {
        if (accounts[i].accountNumber == accountNumber) {
            accounts[i].balance += amount;
            printf("Deposit successful. New balance: %.2lf\n", accounts[i].balance);
            return;
        }
    }
    printf("Account not found.\n");
}

// Function to withdraw money
void withdraw(struct Account accounts[], int numAccounts, int accountNumber, double amount) {
    for (int i = 0; i < numAccounts; i++) {
        if (accounts[i].accountNumber == accountNumber) {
            if (accounts[i].balance >= amount) {
                accounts[i].balance -= amount;
                printf("Withdrawal successful. New balance: %.2lf\n", accounts[i].balance);
            } else {
                printf("Insufficient balance.\n");
            }
            return;
        }
    }
    printf("Account not found.\n");
}

// Function to check account balance
void checkBalance(struct Account accounts[], int numAccounts, int accountNumber) {
    for (int i = 0; i < numAccounts; i++) {
        if (accounts[i].accountNumber == accountNumber) {
            printf("Account Holder: %s\n", accounts[i].name);
            printf("Balance: %.2lf\n", accounts[i].balance);
            return;
        }
    }
    printf("Account not found.\n");
}

int main() {
    struct Account accounts[100];
    int numAccounts = 0;

    int choice;
    do {
        printf("\nBanking Management System\n");
        printf("1. Create Account\n");
        printf("2. Deposit\n");
        printf("3. Withdraw\n");
        printf("4. Check Balance\n");
        printf("5. Exit\n");
        printf("Enter your choice: ");
        scanf("%d", &choice);

        switch (choice) {
            case 1:
                createAccount(accounts, &numAccounts);
                break;
            case 2:
                {
                    int accountNumber;
                    double amount;
                    printf("Enter account number: ");
                    scanf("%d", &accountNumber);
                    printf("Enter amount to deposit: ");
                    scanf("%lf", &amount);
                    deposit(accounts, numAccounts, accountNumber, amount);
                }
                break;
            case 3:
                {
                    int accountNumber;
                    double amount;
                    printf("Enter account number: ");
                    scanf("%d", &accountNumber);
                    printf("Enter amount to withdraw: ");
                    scanf("%lf", &amount);
                    withdraw(accounts, numAccounts, accountNumber, amount);
                }
                break;
            case 4:
                {
                    int accountNumber;
                    printf("Enter account number: ");
                    scanf("%d", &accountNumber);
                    checkBalance(accounts, numAccounts, accountNumber);
                }
                break;
            case 5:
                printf("Goodbye!\n");
                break;
            default:
                printf("Invalid choice. Please try again.\n");
        }
    } while (choice != 5);

    return 0;
}
